bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/BaseThingHandlerFactory.java

```diff
@@ -67,17 +66,13 @@ protected void activate(ComponentContext componentContext) {
     }
 
     /**
-     * Disposes the {@link BaseThingHandlerFactory}. If this method is
-     * overridden by a sub class, the implementing method must call <code>super.deactivate(componentContext)</code>
-     * first.
+     * Disposes the {@link BaseThingHandlerFactory}. If this method is overridden by a sub class, the implementing
+     * method must call <code>super.deactivate(componentContext)</code> first.
      *
      * @param componentContext
      *            component context (must not be null)
      */
     protected void deactivate(ComponentContext componentContext) {
-        for (ServiceRegistration<ThingHandler> serviceRegistration : this.thingHandlers.values()) {
```

Comment:
Hm, isn't dropping the thingHander service registrations a problem here?
If the factory goes down (because some required services are not available anymore), this meant in the past that handlers were also unregistered and hence their Things go into UNINITIALIZED/HANDLER_MISSING status.

Now, it seems that the handlers would be kept, so a binding has no possibility to dispose them anymore?